server/common/annotations.py

```diff
@@ -0,0 +1,231 @@
+from datetime import datetime
+import re
+from uuid import uuid4
+import os
+import pandas as pd
+from hashlib import blake2b
+import base64
+from server import __version__ as cellxgene_version
+import threading
+from server.common.errors import AnnotationsError, OntologyLoadFailure
+from server.common.data_locator import DataLocator
+from server.common.utils import series_to_schema
+import fsspec
+import fastobo
+import traceback  # use built-in formatter for SyntaxError
+from flask import session
+
+
+class Annotations(object):
```

Comment:
this class seems to be missing some required interfaces (abstract) declarations, such as write_labels().

Reply:
fixed.